migration/sql-files/080-remove-unknown-link-types.sql

```diff
@@ -0,0 +1,95 @@
+SET LOCAL linktypes.related = '{{index . 1}}';
+SET LOCAL linktypes.parenting = '{{index . 2}}';
+
+SET LOCAL linkcats.systemcat = '{{index . 3}}';
+SET LOCAL linkcats.usercat = '{{index . 4}}';
+
+-- The following link types exist in the current production database but they
+-- are not known to the code and therefore we re-assign all links associated to
+-- those link types with their appropriate link type and later remove these
+-- unknown link types.
+SET LOCAL linktypes.unknown_bug_blocker = 'aad2a4ad-d601-4104-9804-2c977ca2e0c1';
+SET LOCAL linktypes.unknown_related = '355b647b-adc5-46b3-b297-cc54bc0554e6';
+SET LOCAL linktypes.unknown_parenting = '7479a9b9-8607-46fa-9535-d448fa8768ab';
+
+-- Re-assign any existing link to correct type but avoid those links that
+-- already exist.
+UPDATE work_item_links l 
+    SET link_type_id = current_setting('linktypes.bug_blocker')::uuid 
+    WHERE link_type_id = current_setting('linktypes.unknown_bug_blocker')::uuid
+    AND NOT EXISTS(
+        SELECT * FROM work_item_links
+        WHERE
+            link_type_id = current_setting('linktypes.bug_blocker')::uuid
+            AND source_id = l.source_id
+            AND target_id = l.target_id
+        );
+UPDATE work_item_links l 
+    SET link_type_id = current_setting('linktypes.related')::uuid 
+    WHERE link_type_id = current_setting('linktypes.unknown_related')::uuid
+    AND NOT EXISTS(
+        SELECT * FROM work_item_links
+        WHERE
+            link_type_id = current_setting('linktypes.related')::uuid
+            AND source_id = l.source_id
+            AND target_id = l.target_id
+        );
+UPDATE work_item_links l 
+    SET link_type_id = current_setting('linktypes.parenting')::uuid 
+    WHERE link_type_id = current_setting('linktypes.unknown_parenting')::uuid
+    AND NOT EXISTS(
+        SELECT * FROM work_item_links
+        WHERE
+            link_type_id = current_setting('linktypes.parenting')::uuid
+            AND source_id = l.source_id
+            AND target_id = l.target_id
+        );
+
+-- Update link types in all revision entries
+UPDATE work_item_link_revisions 
+    SET work_item_link_type_id = current_setting('linktypes.bug_blocker')::uuid 
+    WHERE work_item_link_type_id = current_setting('linktypes.unknown_bug_blocker')::uuid;
+UPDATE work_item_link_revisions 
+    SET work_item_link_type_id = current_setting('linktypes.related')::uuid 
+    WHERE work_item_link_type_id = current_setting('linktypes.unknown_related')::uuid;
+UPDATE work_item_link_revisions 
+    SET work_item_link_type_id = current_setting('linktypes.parenting')::uuid 
+    WHERE work_item_link_type_id = current_setting('linktypes.unknown_parenting')::uuid;
+
+-- Update revisions
+UPDATE work_item_link_revisions rev SET
+    work_item_link_type_id = (SELECT link_type_id FROM work_item_links WHERE id = rev.work_item_link_id);
+
+-- Delete all revisions that point to a link that use the old link types 
+DELETE FROM work_item_link_revisions 
```

Comment:
This should not do anything at this point.. remove it just in case?

Reply:
You're right.